helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -249,20 +261,40 @@ private String createAndStartContainer()
   }
 
   private String startContainer(final String image, final Optional<String> dockerVersion)
-      throws InterruptedException, DockerException {
+      throws InterruptedException, DockerException, SecretVolumeException {
 
     // Get container image info
     final ImageInfo imageInfo = docker.inspectImage(image);
     if (imageInfo == null) {
       throw new HeliosRuntimeException("docker inspect image returned null on image " + image);
     }
 
+    final HostConfig.Builder hostConfigBuilder = config.hostConfig(dockerVersion).toBuilder();
+
+    final ImmutableList.Builder<String> binds = ImmutableList.builder();
+    if (hostConfigBuilder.binds() != null) {
+      binds.addAll(hostConfigBuilder.binds());
+    }
+
+    if (secretVolumeManager.isPresent() && config.secretsConfig() != null) {
+      log.info("Secret volume manager enabled for {}", config.containerName());
+      final String source = secretVolumeManager.get()
+          .create(config.containerName(), config.secretsConfig());
```

Comment:
My understanding is that if this call fails `startContainer()` will throw `SecretVolumeException` and the Supervisor will retry.

There's an open question as to what to do if this call fails due to secrets already existing for the requested container name. Currently `secret-volume` will return a 500 and thus this would fail forever until the secrets were manually destroyed (either via the `secret-volume` API or just unmounting them manually).

`secret-volume` could, instead of returning 500, either:
* Just return 200 OK, thus causing whatever secrets exist at the path to be exposed. This could be insecure and lead to stale secret data being exposed.
* Internally destroy the existing secrets at the requested path and attempt to replace them with newly procured secrets.